From 34c25d67c33050f42f0f243c0fb6674c28ff44b5 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@smile.fr>
Date: Sat, 6 Sep 2025 00:31:48 +0200
Subject: [PATCH] remove libasync.o and libbif.o from deps

there are no implicit rules to build them and they get wrong CFLAGS
when compiled with implicit rule.

They are built as part of iozone_linux.o target anyway

Only fixed linux targets for now

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream: Pending
[Romain:
  rebase on 3.508
  convert to git format
]
Signed-off-by: Romain Naour <romain.naour@smile.fr>
---
Patch from Yocto:
https://github.com/openembedded/meta-openembedded/blob/walnascar/meta-oe/recipes-benchmark/iozone3/iozone3/parallelism.patch
---
 src/current/makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/current/makefile b/src/current/makefile
index bd66a4e..ba5759f 100755
--- a/src/current/makefile
+++ b/src/current/makefile
@@ -171,7 +171,7 @@ hpux_no_ansi:	iozone_hpux_no.o  libbif.o
 #
 # GNU 'C' compiler Linux build with threads, largefiles, async I/O 
 #
-linux:	iozone_linux.o libasync.o libbif.o fileop_linux.o pit_server.o
+linux:	iozone_linux.o fileop_linux.o pit_server.o
 	$(CC)  $(CFLAGS) $(LDFLAGS) iozone_linux.o libasync.o libbif.o \
 		-lpthread -lrt -o iozone
 	$(CC)  $(CFLAGS) -Dlinux fileop_linux.o -o fileop
@@ -198,7 +198,7 @@ linux-noth: iozone_linux-noth.o libbif.o fileop_linux.o pit_server.o
 #
 # GNU 'C' compiler Linux build for powerpc chip with threads, largefiles, async I/O 
 #
-linux-powerpc: iozone_linux-powerpc.o  libbif.o libasync.o fileop_linux-ppc.o pit_server.o
+linux-powerpc: iozone_linux-powerpc.o fileop_linux-ppc.o pit_server.o
 	$(CC) -O3 $(LDFLAGS) iozone_linux-powerpc.o libasync.o \
 		libbif.o -lpthread  -lrt -o iozone
 	$(CC)  -O3 -Dlinux fileop_linux-ppc.o -o fileop
@@ -206,7 +206,7 @@ linux-powerpc: iozone_linux-powerpc.o  libbif.o libasync.o fileop_linux-ppc.o pi
 #
 # GNU 'C' compiler Linux build for sparc chip with threads, largefiles, async I/O 
 #
-linux-sparc: iozone_linux-sparc.o  libbif.o libasync.o fileop_linux.o pit_server.o
+linux-sparc: iozone_linux-sparc.o fileop_linux.o pit_server.o
 	$(CC) -O3 $(LDFLAGS) iozone_linux-sparc.o libasync.o libbif.o \
 		-lpthread -lrt -o iozone
 	$(CC) -O3 -Dlinux fileop_linux.o -o fileop
@@ -215,7 +215,7 @@ linux-sparc: iozone_linux-sparc.o  libbif.o libasync.o fileop_linux.o pit_server
 #
 # GNU 'C' compiler Linux build with threads, largefiles, async I/O 
 #
-linux-ia64:	iozone_linux-ia64.o  libbif.o libasync.o fileop_linux-ia64.o pit_server.o
+linux-ia64:	iozone_linux-ia64.o fileop_linux-ia64.o pit_server.o
 	$(CC) -O3 $(LDFLAGS) iozone_linux-ia64.o libbif.o libasync.o \
 		-lrt -lpthread -o iozone
 	$(CC)  -O3 -Dlinux fileop_linux-ia64.o -o fileop
@@ -224,7 +224,7 @@ linux-ia64:	iozone_linux-ia64.o  libbif.o libasync.o fileop_linux-ia64.o pit_ser
 #
 # GNU 'C' compiler Linux build for powerpc chip with threads, largefiles, async I/O 
 #
-linux-powerpc64: iozone_linux-powerpc64.o  libbif.o libasync.o fileop_linux-ppc64.o pit_server-linux-powerpc64.o
+linux-powerpc64: iozone_linux-powerpc64.o fileop_linux-ppc64.o pit_server-linux-powerpc64.o
 	$(CC) -O3 -Dunix -DHAVE_ANSIC_C -DSHARED_MEM -DASYNC_IO \
 		-D_LARGEFILE64_SOURCE -Dlinux \
 		iozone_linux-powerpc64.o libasync.o libbif.o -lpthread \
@@ -235,7 +235,7 @@ linux-powerpc64: iozone_linux-powerpc64.o  libbif.o libasync.o fileop_linux-ppc6
 #
 # GNU 'C' compiler Linux build with threads, largefiles, async I/O
 #
-linux-arm:	iozone_linux-arm.o  libbif.o libasync.o fileop_linux-arm.o pit_server.o
+linux-arm:	iozone_linux-arm.o fileop_linux-arm.o pit_server.o
 	$(CC) -O3 $(LDFLAGS) iozone_linux-arm.o libbif.o libasync.o \
 		-lrt -lpthread -o iozone
 	$(CC) -O3 -Dlinux fileop_linux-arm.o -o fileop
@@ -267,7 +267,7 @@ endif
 #
 # GNU 'C' compiler Linux build with threads, largefiles, async I/O 
 #
-linux-AMD64:	iozone_linux-AMD64.o  libbif.o libasync.o fileop_linux-AMD64.o pit_server.o
+linux-AMD64:	iozone_linux-AMD64.o fileop_linux-AMD64.o pit_server.o
 	$(CC)  -O3 $(LDFLAGS) iozone_linux-AMD64.o libbif.o libasync.o \
 		-lrt -lpthread -o iozone
 	$(CC)  -O3 -Dlinux fileop_linux-AMD64.o -o fileop
@@ -276,7 +276,7 @@ linux-AMD64:	iozone_linux-AMD64.o  libbif.o libasync.o fileop_linux-AMD64.o pit_
 #
 # GNU 'C' compiler Linux build with S/390, threads, largfiles, async I/O
 #
-linux-S390:	iozone_linux-s390.o libbif.o libasync.o fileop_linux-s390.o pit_server.o
+linux-S390:	iozone_linux-s390.o fileop_linux-s390.o pit_server.o
 	$(CC)  -O2 $(LDFLAGS) -lpthread -lrt iozone_linux-s390.o \
 		libbif.o libasync.o -o iozone
 	$(CC)  -O3 -Dlinux fileop_linux-s390.o -o fileop
@@ -285,7 +285,7 @@ linux-S390:	iozone_linux-s390.o libbif.o libasync.o fileop_linux-s390.o pit_serv
 #
 # GNU 'C' compiler Linux build with S/390, threads, largfiles, async I/O
 #
-linux-S390X:	iozone_linux-s390x.o libbif.o libasync.o fileop_linux-s390x.o pit_server.o
+linux-S390X:	iozone_linux-s390x.o fileop_linux-s390x.o pit_server.o
 	$(CC)  -O2 $(LDFLAGS) -lpthread -lrt iozone_linux-s390x.o \
 		libbif.o libasync.o -o iozone
 	$(CC)  -O3 -Dlinux fileop_linux-s390x.o -o fileop
-- 
2.51.0

